tests/test_machine.py

```diff
@@ -0,0 +1,25 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+import os
+
+import pytest
+
+import mkosi.machine as machine
+
+
+@pytest.mark.parametrize("verb", ["qemu", "boot"])
+@pytest.mark.skipif(os.getuid() != 0, reason="Must be invoked as root.")
+@pytest.mark.integration
+def test_run(verb: str) -> None:
+    boot = ["-d", "fedora", "-f", "--autologin"]
+
+    if verb == "qemu":
+        boot += ["--bootable", "--qemu-headless", "--hostonly-initrd", "--network-veth", "--ssh", verb]
```

Comment:
Having to add these arguments for every test would be rather repetitive and error prone no? The Machine class should make sure to add the arguments it needs itself.